### PR TITLE
Fixes #20908 - postpone initialization of Dynflow runtime

### PR DESCRIPTION
**Pull request by iNecas**
In #18618, the initialization code was moved from foreman-tasks to
foreman. There was however one thing done differently, where the
initialization of dynflow world was moved directly to the time
`Rails.application.dynflow` was called. This is not right, as there are
situations where plugins go and call `Rails.application.dynflow.config`
before the whole stack is prepared, which leads to database connection
being opened too early, which can cause the whole lot of issues (as
we've hit currently with Katello).

There was also change in the signature of `Dynflow::Rails#initialize`
method, that needs to be updated.

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -246,12 +246,27 @@ def dynflow
         if defined?(ForemanTasks)
           ForemanTasks.dynflow
         else
-          Dynflow::Rails.new(Foreman::Dynflow::Configuration.new)
+          ::Dynflow::Rails.new(nil, Foreman::Dynflow::Configuration.new)
```
Comment (iNecas) [inline config/application.rb:249]: `Dynflow::Rails` was resolving to `Foreman::Dynlfow::Rails` which was not existing: the after_initialize block actually discovered this issue as it seems this code-path was not run in tests before.

**Comment by iNecas** [comment]
And all the tests are green again :)

File: config/application.rb (hunk 1 above)
Comment (dLobatog) [reply config/application.rb:249]: Thanks, I had to do something similar in https://github.com/theforeman/foreman/pull/4717/, can confirm it fixes the problem

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
Thanks @iNecas - this should go into 1.16 I suppose?

**Comment by iNecas** [comment]
yes please

**Comment by dLobatog** [comment]
Cherry-picked as 7ad59f2a3048f8f81d6221f7ec87bd48d478110b

File: config/application.rb (hunk 1 above)
Comment (iNecas) [reply config/application.rb:249]: Yes, this should fix the issue you were facing there + making it work with later dynflow version as well that came with https://github.com/Dynflow/dynflow/commit/ff129a6b92eb209a82e51d177034b6f9187bff19#diff-56fbff8c6eb846b55588d78db914924aR10

**Pull request merged**